components/wysiwyg/wysiwyg.tsx

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {KeyboardShortcutCommand} from '@tiptap/core';
+import React, {useEffect} from 'react';
+import type {FormEvent} from 'react';
+import styled from 'styled-components';
+
+import {
+    EditorContent,
+    useEditor,
+} from '@tiptap/react';
+import type {JSONContent} from '@tiptap/react';
+
+// load all highlight.js languages
+import {lowlight} from 'lowlight';
+
+import type {NewPostDraft} from 'types/store/draft';
+
+import EmojiPicker from './components/emoji-picker';
+import {Extensions, SuggestionConfig} from './extensions';
+import {htmlToMarkdown} from './utils/toMarkdown';
+
+// import all custom components, extensions, etc.
+import Toolbar from './components/toolbar';
+import SendButton from './components/send-button';
+
+import {PropsFromRedux} from './index';
+
+const WysiwygContainer = styled.div`
+    margin: 0 24px 24px;
+    border: 2px solid rgba(var(--center-channel-color-rgb), 0.32);
+    border-radius: 4px;
+`;
+
+const EditorContainer = styled.div`
```

Comment:
Would it make sense to move these into their own files?

Reply:
I was thinking about that as well ... something like `wysiwyg.components.tsx` or similar would do